db/migrate/20180704201722_add_theme_id_to_user_option.rb

```diff
@@ -0,0 +1,45 @@
+class AddThemeIdToUserOption < ActiveRecord::Migration[5.2]
+  def up
+    add_column :user_options, :theme_ids, :integer, array: true,  null: false, default: []
+
+    execute(
+      "UPDATE user_options AS uo
+       SET theme_ids = (
+         SELECT array_agg(themes.id)
+         FROM themes
+         INNER JOIN user_options
+         ON themes.key = user_options.theme_key
+         WHERE user_options.user_id = uo.user_id
+       ) WHERE uo.theme_key IN (SELECT key FROM themes)"
+    )
+
+    begin
+      Migration::SafeMigrate.disable!
+      remove_column :user_options, :theme_key
+    ensure
+      Migration::SafeMigrate.enable!
+    end
+  end
+
+  def down
+    add_column :user_options, :theme_key, :string
```

Comment:
less concerned about down migrations, you can just throw an irreversible migration exception here